### PR TITLE
[DOT-1] added a function to speed up basic git tasks

### PR DESCRIPTION
**Pull request by RonitRudra**
This is a simple alias added with a lot of functions to automate some frequent git commands. I took the liberty of christening the branches as DOT-<number>
This is more of a crude POC than a finished script. For now it's inside the dbuy.zsh_aliases but could be moved to a script file of its own to make things more modular and clean.

**FEATURES**:
 + Commit all changes in your local branch by using *gme -c "commit message"* This would add all files and create a commit with the message *[DBUY-0001] commit message* if your branch name was DBUY-0001
+ Switch to an existing branch with *gme -b BRANCHNAME*
+ Rebase all local changes by kicking off interactive rebase using *gme -r*. This adds all local commits to the rebase so if your branch has 30 commits in it, it will rebase 30. Does away with manually counting the commit numbers when doing a rebase.
+ Rebase against the dev branch using *gme -R*. Takes care of switching to dev, pulling, switching to the branch you were working on and then rebasing against origin/dev
+ See usage and help using *gme -h*

**more to come soon**

### CODE REVIEW

File: zsh/dbuy.zsh_aliases [hunk 1]
```diff
@@ -26,3 +26,89 @@ function find_mail_errors() {
...
+function git_made_easy(){
+    local CURRENT_BRANCH
+    local COMMIT_MESSAGE
+    local -A ARGERROR
+    local -A ARGDETAIL
+    ARGDETAIL=(c "Commit all files to current branch" r "Rebase all local changes" b "Checkout to a local branch" R "Rebase current branch with dev")
+    ARGERROR=(c "Commit Message" b "Branch Name")
+    CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+    function commit_all() {
+        if [ "$CURRENT_BRANCH" = "dev" ]; then
```
Comment (2ps) [inline zsh/dbuy.zsh_aliases]: Should check for `staging`  and `master` as well.  Go ahead and switch to `[[` and `]]` around the if and use `==`.

File: zsh/dbuy.zsh_aliases [hunk 2]
```diff
@@ -26,3 +26,89 @@ function find_mail_errors() {
...
+            echo "You are on dev. ABORTING REBASE!"
+        else
+            local NUM
+            NUM=$(git log --oneline dev.. | wc -l)
+            echo "Rebasing $NUM commits on $CURRENT_BRANCH"
+            git rebase -i HEAD~${NUM}
+        fi
+    }
+    function rebase_dev() {
+        if [ "$CURRENT_BRANCH" = "dev" ]; then
```
Comment (2ps) [inline zsh/dbuy.zsh_aliases]: Or `staging` or `master`.

**Comment by ghost** [comment]
This seems pretty nifty. One note, if I am reading code correctly, the rebase is predicated on 100% adherence to naming standard of commits having "[branch_name]" right?

If so, should there be some fuzziness around the brackets?

**Comment by RonitRudra** [comment]
> This seems pretty nifty. One note, if I am reading code correctly, the rebase is predicated on 100% adherence to naming standard of commits having "[branch_name]" right?
> 
> If so, should there be some fuzziness around the brackets?

@blissrage the rebase is independent of the way you write your commit messages as it only looks at the log of the current branch.

**Comment by RonitRudra** [comment]
@blissrage but yeah it only works for branches created off *dev* since I hard-coded that in there.
So dev --> DBUY-xxxx would work but not DBUY-xxxx -> DBUY-xxxx-child as the code now needs the parent of DBUY-xxxx-child.

**Review by ghost — APPROVED** [review]

**Comment by 2ps** [comment]
@RonitRudra :  bump--still have comments to address.

**Comment by RonitRudra** [comment]
@2ps What will had mentioned? The rebase does not depend on naming standard as it gets the commits only from the current branch. Using the standard is good practice but even without it the rebase should work normally.

**Review by 2ps — CHANGES_REQUESTED** [review]

**Comment by RonitRudra** [comment]
@2ps made the changes that you asked for -> `[[ ==` as well as checking for staging and master branches . (rebasing the commits and pushing gets rid of everyone's change requests though)

**Pull request merged**